apps/framework-cli/src/cli/local_webserver.rs

```diff
@@ -211,31 +211,55 @@ async fn get_consumption_api_res(
     debug!("Creating client for route: {:?}", url);
     {
         let consumption_apis = consumption_apis.read().await;
+
         let consumption_name = req
             .uri()
             .path()
             .strip_prefix("/consumption/")
             .unwrap_or(req.uri().path());
 
+        // Check for exact match first
         if !consumption_apis.contains(consumption_name) {
-            if !is_prod {
-                use crossterm::{execute, style::Print};
-                let msg = format!(
-                    "Consumption API {} not found. Available consumption paths: {}",
-                    consumption_name,
-                    consumption_apis
-                        .iter()
-                        .map(|p| p.as_str())
-                        .collect::<Vec<&str>>()
-                        .join(", ")
-                );
-                let _ = execute!(std::io::stdout(), Print(msg + "\n"));
+            // Try to find a versioned match
+            let mut found_match = false;
+
+            // Check if this looks like a versioned request (contains a slash)
+            if consumption_name.contains('/') {
+                // Split the path to get base name and potential version
+                let parts: Vec<&str> = consumption_name.split('/').collect();
+                if parts.len() >= 2 {
+                    let base_name = parts[0];
+
+                    // Look for any API that starts with the base name followed by a slash
+                    for api in consumption_apis.iter() {
```

Comment:
To be honest - I am not in love with this `for loop` but I am not sure I have a much better solution from an implementation perspective.

For every request, we will have to find the consumption API that fits. If it is the latest in a 1000 endpoint list then it will have to go to the end. 

In high load scenarios that will eat a bunch of time. The previous lookup was O(1) so really fast.

I am tempted to remove all this code that's in the end just there to return an early 404 and let the node / python service return the 404.

What do you think?